contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,287 @@
+         * of 32ETH have been staked.
+         * But someone could mess with our strategy by sending some WETH to it. And we might want to deposit just
+         * enough WETH to add it up to 32 so it can be staked. For that reason the check is left out.
+         *
+         * WETH sitting on the strategy won't interfere with the accounting since accounting only operates on ETH.
+         */
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /// @notice Deposit the entire balance of WETH asset in the strategy into the underlying platform
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 wethBalance = IERC20(WETH_TOKEN_ADDRESS).balanceOf(
+            address(this)
+        );
+        if (wethBalance > 0) {
+            _deposit(WETH_TOKEN_ADDRESS, wethBalance);
+        }
+    }
+
+    /// @notice Withdraw WETH from this contract. Used only if some WETH for is lingering on the contract. That
+    /// can happen when:
+    ///   - the deposit was not a multiple of 32 WETH
+    ///   - someone sent WETH directly to this contract
+    /// @param _recipient Address to receive withdrawn assets
+    /// @param _asset WETH to withdraw
+    /// @param _amount Amount of WETH to withdraw
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        _withdraw(_recipient, _asset, _amount);
+    }
+
+    function _withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        if (_recipient == address(0)) {
+            revert EmptyRecipient();
+        }
+
+        emit Withdrawal(_asset, address(0), _amount);
+        IERC20(_asset).safeTransfer(_recipient, _amount);
+    }
+
+    /// @notice Remove all supported assets from the underlying platform and send them to Vault contract.
+    function withdrawAll() external override onlyVaultOrGovernor nonReentrant {
+        uint256 wethBalance = IERC20(WETH_TOKEN_ADDRESS).balanceOf(
+            address(this)
+        );
+        if (wethBalance > 0) {
+            _withdraw(vaultAddress, WETH_TOKEN_ADDRESS, wethBalance);
+        }
+    }
+
+    function _abstractSetPToken(address _asset, address) internal override {}
+
+    /// @notice Returns the total value of (W)ETH that is staked to the validators
+    /// and also present on the native staking and fee accumulator contracts
+    /// @param _asset      Address of weth asset
+    /// @return balance    Total value of (W)ETH 
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256 balance)
+    {
+        if (_asset != WETH_TOKEN_ADDRESS) {
+            revert NotWeth();
+        }
+
+        balance = activeDepositedValidators * 32 ether;
+        balance += beaconChainRewardWETH;
+        balance += FEE_ACCUMULATOR_ADDRESS.balance;
```

Comment:
I don't think we want to include the MEV rewards in the FeeAccumulator if those are going to be harvested and sent to through the dripper.